apps/resource_manager/lib/identities/commands/inputs/create_user.ex

```diff
@@ -5,40 +5,63 @@ defmodule ResourceManager.Identities.Commands.Inputs.CreateUser do
 
   use ResourceManager.Input
 
+  alias ResourceManager.Credentials.Schemas.Password
   alias ResourceManager.Identities.Schemas.User
 
   @typedoc "Create user input fields"
   @type t :: %__MODULE__{
           username: String.t(),
           status: String.t(),
-          password_hash: String.t(),
-          password_algorithm: String.t(),
-          scopes: list(String.t()) | nil
+          password: %{
+            value: String.t(),
+            algorithm: String.t() | nil,
+            salt: integer() | nil
+          },
+          permission: %{
+            scopes: list(String.t()) | nil
+          }
         }
 
-  @required [:username, :password_hash, :password_algorithm]
-  @optional [:scopes]
+  @required [:username]
+  @optional [:status]
   embedded_schema do
     # Identity
     field :username, :string
     field :status, :string, default: "active"
 
-    # Credentials
-    field :password_hash, :string
-    field :password_algorithm, :string, default: "argon2"
+    embeds_one :password, Credential, primary_key: false do
+      field :value, :string
```

Comment:
```suggestion
      field :hash, :string
```

Reply:
In this case it is the password value and not it's hash.